securitas/tests/unit/controller/test_password.py

```diff
@@ -9,3 +11,117 @@ def test_password_reset(client):
     page = BeautifulSoup(result.data, 'html.parser')
     assert page.title
     assert page.title.string == 'Password Reset - The Fedora Project'
+
+
+def test_missing_credentials(client):
+    """Verify that missing details are caught"""
+    result = client.post(
+        '/password-reset',
+        data={
+            "username": "",
+            "current_password": "",
+            "password": "",
+            "password_confirm": "",
+        },
+        follow_redirects=True,
+    )
+    assert result.status_code == 200
+    page = BeautifulSoup(result.data, 'html.parser')
+    messages = page.select(".flash-messages .red")
+    assert len(messages) == 1
+    assert messages[0].get_text(strip=True) == 'Please fill in all fields to reset your password'
```

Comment:
There are other ways to check for flash messages, you can for example look at `test_user.test_user_edit_post`. The idea is to not follow redirects, and use `get_flashed_messages()` to introspect which flash messages have been generated.
I think it's better because it does not depend on the HTML produced to display them (which may change when we work on Ryan's designs).

Reply:
One more thing, the recommended way of handling form erros (in WTForms) is to not redirect on errors but let the `form.errors` dict be filled with the errors and render them. This way you can display errors next to the field that is causing them, and the form fields will still be filled with the values you entered previously (except passwords).

I think you should convert the `password` controller to work that way, you can check out the changes I've made to the `registration` controller for an example. When that's done you'll be able to check for errors in the tests without using the flash messages. There again you can look at the corresponding tests in `test_registration.py`.